### DISCUSSION TITLE
データベースの移行

### DISCUSSION THREAD

**Discussion by nattyan-tv** [team-i2021/nira_bot#30] [General]
HTTP_dbが完成し次第、データベースマネージャーを作って色々テストしたら、HTTP_dbで本実装する（予定）
[nattyan-tv/HTTP_db](https://github.com/nattyan-tv/HTTP_db)

---

サーバーのデータベースから毎回データを取るから、遅い。
一応エラーが出ても何とかなるっちゃ何とかなるようにしたけども（例えば`user_join.py`とか）...
毎度Googleサーバーに接続しない何か別の方法を考えましょうか。

[とあるCommit](https://github.com/team-i2021/nira_bot/commit/10673b6cf42f2ace6ced4a5c2fa5ebb38afb00c3)でやった、エラー処理のやつが功を奏して、ちゃんと「送信は」してくれるようになった。

**Comment by nattyan-tv** [comment]
# そのまま推奨の変数形式に落とせないやつ
（このレポジトリじゃない気もしたけど...()）
ここでいう推奨とは
```py
{nextcord.Guild.id: {nextcord.abc.GuildChannel.id: {any} } }
```
みたいな、`dict[int, dict[int, any]]`の形式...
つまり、以下のデータはそう簡単にはこの形には落とせない、通常とは別の形の変数。
## PickleData
- force_ss_list
  ```py
  {nextcord.Guild.id: [nextcord.TextChannel.id, nextcord.Message.id] }
  ```
- PollViews
  ```py
  [
    [option1, option2],
    [option1, option2, option3]
  ]
  ```
- PersistentViews
  ```py
  [
    [
      [1, nextcord.Role.id],
      [2, nextcord.Role.id]
    ],
    [
      [1, nextcord.Role.id],
      [2, nextcord.Role.id]
    ]
  ]
  ```
- welcome_id_list
  ```py
  {nextcord.Guild.id: nextcord.TextChannel.id}
  ```
- steam_server_list
  ```py
  {nextcord.Guild.id: {"value": 1, "1_ad": ("localhost", 27015), "1_nm": "TestServer"} }
  ```
- role_keeper
  ```py
  {nextcord.Guild.id: {"rk": 1, nextcord.Member.id: [nextcord.Role.id] } }
  ```
- mc_server_list
  ```py
  {nextcord.Guild.id: {"value": 2, 1: ["TestJavaServer", "localhost:25565", "java"], 2: ["TestBEServer", "localhost:19132", "be"] } }
  ```
## Google Spread Sheet
- Dissokuデータ
  ```py
  {nextcord.Guild.id: nextcord.Role.id}
  ```
- Remindデータ[OK]
  ```py
  {nextcord.Guild.id: {"hh:mm": "Remind content"} }
  ```

基本的には大規模メンテナンスの時に全部のデータを移行するから、その時に全部手動で移行させるときのためのメモとして...()
あとは、移行するに伴って、データの保存形式も変更したい気がするからそのためのメモとしても

### CheckList
- [ ] ~~amuse~~
- [x] autorole
- [x] bump
- [ ] ~~captcha~~
- [ ] ~~check~~
- [ ] ~~code~~
- [ ] ~~debug~~
- [ ] ~~embed~~
- [ ] ~~get_reaction~~
- [ ] ~~hoyo~~
- [ ] ~~info~~
- [x] invite
- [x] message_dm
- [x] message_role
- [ ] ~~MessageCommand~~
- [x] minecraft
- [x] mod
- [ ] ~~music~~
- [ ] normal_reaction
- [ ] ~~not_ready~~
- [ ] ~~on_error~~
- [x] pin
- [ ] ~~ping~~
- [ ] pollpanel
- [x] reaction
- [x] remind
- [ ] rolepanel
- [x] server_status
- [x] siritori
- [ ] ~~translate~~
- [x] tts
- [x] up
- [x] user_join
- [x] user
- [x] welcome

# Reminder
- [x] コマンドチェック
- [x] PR:ブランチのマージ
- [x] データの移行

**Comment by CrystalAlpha358** [comment]
どうしてHTTP_db側で対処しようとしてるの？データベースはプログラムで使われてる構造をそのまま保存するものではないでしょ
保存するデータ構造を変えること考えた方がいいんじゃない？

**Reply by nattyan-tv** [reply]
HTTP_dbって書いたのがあれだった
保存するデータの構造を変えるためのメモ

**Comment by CrystalAlpha358** [comment]
開きっぱなしになってたことに今更気づいたのでclose...
